vllm/config.py

```diff
@@ -1050,7 +1055,11 @@ def _verify_args(self) -> None:
                 "file an issue with detailed information.")
 
     @property
-    def is_multi_step(self) -> bool:
+    def user_is_multi_step(self) -> bool:
```

Comment:
I agree & I initially tried a local implementation, the problem is that is_multi_step forms a term in so many conditionals in so many methods, it becomes a very big change in order to accomplish something do small ( as you said it is not easy)

Perhaps I can give the different attributes more different names?

Reply:
Update: made the names maximally different.
* `is_multi_step` => `current_step_is_multi_step`
* `user_is_multi_step` => `engine_permits_multi_step_scheduling`

I am strongly inclined to maintain this approach (rather than a "more local approach") since (1) as you say we will not have this problem in the future, so I am not inclined to greatly change the structure of the code in order to set up a more local solution, however (2) we are not at that point in the future yet where best_of > 1 does not matter & there is near-term interest in unblocking customers by hardening the current vLLM version so that it does not crash for multi-step + best_of > 1, and also (3) while the current "non-local" solution I have implemented is maybe not ideal, there is precedent in the codebase for creating "non-local" attributes of this sort, i.e. `Sampler.include_gpu_probs_tensor` which is conditionally set in execute_model() but is not local to execute_model()